MongoDBConnectorCappedCollectionConsumerTest is flaky

## This is a...

<pre><code>
[ ] Feature request
[ ] Regression (a behavior that used to work and stopped working in a new release)
[x] Bug report  <!-- Please search GitHub for a similar issue or PR before submitting -->
[ ] Documentation issue or request
</code></pre>

## Description

[Daily release failed](
https://ci.fabric8.io/blue/organizations/jenkins/syndesis-release-daily/detail/syndesis-release-daily/652/pipeline) because of failing test in MongoDBConnectorCappedCollectionConsumerTest.

Still an issue according to the [daily release failure](https://ci.fabric8.io/view/syndesis/job/syndesis-release-daily/lastFailedBuild/):

```
[ERROR] Failures: 
[ERROR]   MongoDBConnectorCappedCollectionConsumerTest.repeatMongoTest:90->mongoTest:84 mock://result Received message count. Expected: <1> but was: <2>
```

--- a/app/connector/mongodb/src/test/java/io/syndesis/connector/mongo/MongoDBConnectorCappedCollectionConsumerAllOptionsTest.java
+++ b/app/connector/mongodb/src/test/java/io/syndesis/connector/mongo/MongoDBConnectorCappedCollectionConsumerAllOptionsTest.java
@@ -18,7 +18,6 @@ package io.syndesis.connector.mongo;
 import java.io.IOException;
 import java.util.ArrayList;
 import java.util.List;
-import java.util.concurrent.TimeUnit;
 
 import com.fasterxml.jackson.databind.JsonNode;
 import com.mongodb.client.model.CreateCollectionOptions;
@@ -54,13 +53,8 @@ public class MongoDBConnectorCappedCollectionConsumerAllOptionsTest extends Mong
     @AfterClass
     public static void testTrackingIdValue() {
         List<Document> docsFound = database.getCollection("tracking").find().into(new ArrayList<>());
-        System.out.println("Docs found" + docsFound);
         assertEquals(25, (int) docsFound.get(0).getInteger("someTrackingId"));
     }
-
-    // **************************
-    // Tests
-    // **************************
 
     @Override
     protected List<Step> createSteps() {
@@ -73,6 +67,8 @@ public class MongoDBConnectorCappedCollectionConsumerAllOptionsTest extends Mong
     public void mongoTest() throws Exception {
         // When
         MockEndpoint mock = getMockEndpoint("mock:result");
+        // We just retain last message
+        mock.setRetainLast(1);
         mock.expectedMessageCount(3);
         mock.expectedMessagesMatches((Exchange e) -> {
             try {
@@ -81,6 +77,7 @@ public class MongoDBConnectorCappedCollectionConsumerAllOptionsTest extends Mong
                 List<String> doc = e.getMessage().getBody(List.class);
                 JsonNode jsonNode = MAPPER.readTree(doc.get(0));
                 Assertions.assertThat(jsonNode).isNotNull();
+                Assertions.assertThat(jsonNode.get("someTrackingId").asInt()).isEqualTo(25);
             } catch (IOException ex) {
                 return false;
             }
@@ -101,6 +98,6 @@ public class MongoDBConnectorCappedCollectionConsumerAllOptionsTest extends Mong
         collection.insertOne(doc3);
 
         // Then
-        MockEndpoint.assertIsSatisfied(5, TimeUnit.SECONDS, mock);
+        mock.assertIsSatisfied();
     }
 }

--- a/app/connector/mongodb/src/test/java/io/syndesis/connector/mongo/MongoDBConnectorCappedCollectionConsumerTest.java
+++ b/app/connector/mongodb/src/test/java/io/syndesis/connector/mongo/MongoDBConnectorCappedCollectionConsumerTest.java
@@ -17,13 +17,14 @@ package io.syndesis.connector.mongo;
 
 import java.io.IOException;
 import java.util.List;
-import java.util.concurrent.TimeUnit;
+import java.util.UUID;
 
 import com.fasterxml.jackson.databind.JsonNode;
 import com.mongodb.client.model.CreateCollectionOptions;
 import io.syndesis.common.model.integration.Step;
 import org.apache.camel.Exchange;
 import org.apache.camel.component.mock.MockEndpoint;
+import org.assertj.core.api.Assertions;
 import org.bson.Document;
 import org.junit.BeforeClass;
 import org.junit.Test;
@@ -33,19 +34,15 @@ import org.slf4j.LoggerFactory;
 public class MongoDBConnectorCappedCollectionConsumerTest extends MongoDBConnectorTestSupport {
 
     private static final Logger LOG = LoggerFactory.getLogger(MongoDBConnectorCappedCollectionConsumerTest.class);
-    private static int globalId = 1;
-
-    // **************************
-    // Set up
-    // **************************
+    private static int globalId = 0;
 
     // JUnit will execute this method after the @BeforeClass of the superclass
     @BeforeClass
     public static void doCollectionSetup() {
         // The feature only works with capped collections!
         CreateCollectionOptions opts = new CreateCollectionOptions().capped(true).sizeInBytes(1024 * 1024);
-        database.createCollection("test", opts);
-        LOG.debug("Created a capped collection named test");
+        database.createCollection(COLLECTION, opts);
+        LOG.debug("Created a capped collection named {}", COLLECTION);
     }
 
     @Override
@@ -54,39 +51,39 @@ public class MongoDBConnectorCappedCollectionConsumerTest extends MongoDBConnect
             "id");
     }
 
-    // **************************
-    // Tests
-    // **************************
-
     @Test
     public void mongoTest() throws Exception {
         // When
+        String unique = UUID.randomUUID().toString();
+        int id = globalId++;
         MockEndpoint mock = getMockEndpoint("mock:result");
+        // We just retain last message
+        mock.setRetainLast(1);
         mock.expectedMessageCount(1);
         mock.expectedMessagesMatches((Exchange e) -> {
             try {
                 @SuppressWarnings("unchecked")
                 List<String> doc = e.getMessage().getBody(List.class);
                 JsonNode jsonNode = MAPPER.readTree(doc.get(0));
-                int id = jsonNode.get("id").asInt();
-                String value = jsonNode.get("someKey").asText();
-                return id <= globalId && "someValue".equals(value);
+                Assertions.assertThat(jsonNode.get("unique").asText()).isEqualTo(unique);
+                Assertions.assertThat(jsonNode.get("id").asInt()).isEqualTo(id);
+                return true;
             } catch (IOException ex) {
                 return false;
             }
         });
         // Given
         Document doc = new Document();
-        doc.append("id", globalId++);
-        doc.append("someKey", "someValue");
+        doc.append("id", id);
+        doc.append("unique", unique);
         collection.insertOne(doc);
         // Then
-        MockEndpoint.assertIsSatisfied(5, TimeUnit.SECONDS, mock);
+        mock.assertIsSatisfied();
     }
 
     @Test
     public void repeatMongoTest() throws Exception {
-        // As we are tracking _id, any new insert should trigger the new document only
+        // As we are tracking id, any new insert should trigger the new document only
         mongoTest();
     }
 

--- a/app/connector/mongodb/src/test/java/io/syndesis/connector/mongo/MongoDBConnectorChangeStreamConsumerTest.java
+++ b/app/connector/mongodb/src/test/java/io/syndesis/connector/mongo/MongoDBConnectorChangeStreamConsumerTest.java
@@ -15,9 +15,10 @@
  */
 package io.syndesis.connector.mongo;
 
+import java.io.IOException;
 import java.util.List;
-import java.util.concurrent.TimeUnit;
 
+import com.fasterxml.jackson.databind.JsonNode;
 import com.mongodb.BasicDBList;
 import com.mongodb.MongoClient;
 import de.flapdoodle.embed.mongo.MongodExecutable;
@@ -28,7 +29,9 @@ import de.flapdoodle.embed.mongo.config.Net;
 import de.flapdoodle.embed.mongo.config.Storage;
 import de.flapdoodle.embed.mongo.distribution.Version;
 import io.syndesis.common.model.integration.Step;
+import org.apache.camel.Exchange;
 import org.apache.camel.component.mock.MockEndpoint;
+import org.assertj.core.api.Assertions;
 import org.bson.Document;
 import org.junit.AfterClass;
 import org.junit.BeforeClass;
@@ -68,8 +71,6 @@ public class MongoDBConnectorChangeStreamConsumerTest extends MongoDBConnectorTe
 
         initCluster();
         initClient();
-        //Create a collection needed by this test
-        database.createCollection(COLLECTION);
     }
 
     private static void initCluster() {
@@ -84,6 +85,8 @@ public class MongoDBConnectorChangeStreamConsumerTest extends MongoDBConnectorTe
         config.put("members", members);
         mongoClient.getDatabase("admin").runCommand(new Document("replSetInitiate", config));
         database = mongoClient.getDatabase(DATABASE);
+        //Create a collection needed by this test
+        database.createCollection(COLLECTION);
         collection = database.getCollection(COLLECTION);
     }
 
@@ -98,7 +101,21 @@ public class MongoDBConnectorChangeStreamConsumerTest extends MongoDBConnectorTe
     public void singleInsertTest() throws Exception {
         // When
         MockEndpoint mock = getMockEndpoint("mock:result");
+        mock.setRetainLast(1);
         mock.expectedMessageCount(2);
+        mock.expectedMessagesMatches((Exchange e) -> {
+            try {
+                // We just want to validate the output is coming as json well format
+                @SuppressWarnings("unchecked")
+                List<String> doc = e.getMessage().getBody(List.class);
+                JsonNode jsonNode = MAPPER.readTree(doc.get(0));
+                Assertions.assertThat(jsonNode).isNotNull();
+                Assertions.assertThat(jsonNode.get("test").asText()).isEqualTo("junit2");
+            } catch (IOException ex) {
+                return false;
+            }
+            return true;
+        });
         // Given
         Document doc = new Document();
         doc.append("someKey", "someValue");
@@ -109,6 +126,6 @@ public class MongoDBConnectorChangeStreamConsumerTest extends MongoDBConnectorTe
         doc2.append("test", "junit2");
         collection.insertOne(doc2);
         // Then
-        MockEndpoint.assertIsSatisfied(5, TimeUnit.SECONDS, mock);
+        mock.assertIsSatisfied();
     }
 }

--- a/app/connector/mongodb/src/test/java/io/syndesis/connector/mongo/MongoDBConnectorTestSupport.java
+++ b/app/connector/mongodb/src/test/java/io/syndesis/connector/mongo/MongoDBConnectorTestSupport.java
@@ -123,7 +123,7 @@ public abstract class MongoDBConnectorTestSupport extends ConnectorTestSupport {
 
     protected List<Step> fromMongoTailToMock(String mock, String connector, String db, String collection,
                                          String tailTrackIncreasingField) {
-        return this.fromMongoTailToMock(mock, connector, db, collection, tailTrackIncreasingField, null, null, null, null, null);
+        return this.fromMongoTailToMock(mock, connector, db, collection, tailTrackIncreasingField, true, null, null, null, null);
     }
 
     protected List<Step> fromMongoTailToMock(String mock, String connector, String db, String collection,

